app/routes/review_offer.js

```diff
@@ -45,28 +61,23 @@ export default AuthorizeRoute.extend({
     controller.set("displayOfferOptions", false);
     controller.set("displayCompleteReceivePopup", false);
 
-    if(this.get('backLinkPath') !== null) {
-      controller.set('backLinkPath', this.get('backLinkPath'));
+    if (this.get("backLinkPath") !== null) {
+      controller.set("backLinkPath", this.get("backLinkPath"));
     } else {
-      controller.set('backLinkPath', this.getBackLinkPath(model));
+      controller.set("backLinkPath", this.getBackLinkPath(model));
     }
   },
 
-  getBackLinkPath(offer) {
-    if(offer.get("isSubmitted")) { return "offers"; }
-    else if(offer.get("isReceiving")) { return "offers.receiving"; }
-    else if(offer.get("isReviewed")) { return "in_progress.reviewed"; }
-    else if(offer.get("isUnderReview")) { return "in_progress.reviewing"; }
-    else if(offer.get("isClosed") || offer.get("isCancelled")) {
-      return "finished.cancelled"; }
-    else if(offer.get("isReceived")) { return "finished.received"; }
-    else if(offer.get("isInactive")) { return "finished.inactive"; }
-    else if(offer.get("isScheduled")) {
-      if(offer.get("delivery.isGogovan")) { return "scheduled.gogovan"; }
-      else if(offer.get("delivery.isDropOff")) { return "scheduled.other_delivery"; }
-      else if(offer.get("delivery.isAlternate")) { return "scheduled.collection"; }
-    else { return "offers"; }
+  getBackLinkPath(offer, mapping = BACKLINK_CONDITIONS) {
```

Comment:
FYI: I refactored this because of code-climate. This feels like a hacky navigation, but I didn't want to remove this without knowing the consequences.

Reply:
👍